src/v2/guide/reactivity.md

```diff
@@ -20,7 +20,11 @@ Every component instance has a corresponding **watcher** instance, which records
 
 ## Change Detection Caveats
 
-Due to the limitations of modern JavaScript (and the abandonment of `Object.observe`), Vue **cannot detect property addition or deletion**. Since Vue performs the getter/setter conversion process during instance initialization, a property must be present in the `data` object in order for Vue to convert it and make it reactive. For example:
+Due to limitations in JavaScript, there are types of changes that Vue **cannot detect**. However, there are ways to circumvent them to preserve reactivity.
+
+### With Objects
+
+Vue **cannot detect property addition or deletion**. Since Vue performs the getter/setter conversion process during instance initialization, a property must be present in the `data` object in order for Vue to convert it and make it reactive. For example:
```

Comment:
Since we have emphasized that Vue `**cannot**` in the leading paragraph (L23), emphasizing it again (L27) and again (L62) sounds pretty dramatic IMHO 😅.

Reply:
Yep! I'll also remove the redundant "Due to limitations in JavaScript" in the **For Arrays** sub-section.